src/main/java/space/obminyashka/items_exchange/service/impl/SendGridService.java

```diff
@@ -42,31 +52,40 @@ public class SendGridService implements MailService {
     private int numberOfDaysToKeepDeletedEmails;
 
     @Override
-    public void sendMail(String emailTo, EmailType subject, UUID codeId, String host) throws IOException {
+    public UUID sendEmailTemplateAndGenerateConfrimationCode(String emailTo, EmailType emailType, String host){
         var mail2send = new Mail();
         mail2send.setFrom(sender);
-        mail2send.setTemplateId(subject.template);
+        mail2send.setTemplateId(emailType.template);
 
-        final var personalization = populatePersonalization(subject.topic, codeId, host);
+        UUID codeId = UUID.randomUUID();
+        final var personalization = createPersonalizationAndSetParameters(emailType, codeId, host);
         personalization.addTo(new Email(emailTo));
         mail2send.addPersonalization(personalization);
 
-        var request = createMailRequest(mail2send);
-        final var response = sendGrid.api(request);
-        final var statusCode = response.getStatusCode();
-        log.debug("[SendGridService] A sent email result. STATUS: {} BODY: {}", statusCode, response.getBody());
+        try {
+            Request request  = createMailRequest(mail2send);
+            final var response = sendGrid.api(request);
+            final var statusCode = response.getStatusCode();
+
+            log.debug("[SendGridService] A sent email result. STATUS: {} BODY: {}", statusCode, response.getBody());
+        } catch (IOException e) {
+            log.error("Error while sending {emailType.name()} email", e);
```

Comment:
That is not Kotlin or JS where you can invoke method. You should do it differently: single brackets {} and adding parameter to the log method